src/common/hist_util.cc

```diff
@@ -303,7 +303,9 @@ void DenseCuts::Init
   }
   CHECK_EQ(summary_array.size(), in_sketchs->size());
   size_t nbytes = WXQSketch::SummaryContainer::CalcMemCost(max_num_bins * kFactor);
-  sreducer.Allreduce(dmlc::BeginPtr(summary_array), nbytes, summary_array.size());
+  // TODO(chenqin): after remove is_bootstrap, hist break rabit assumeption
```

Comment:
So what's the thing needs to be done in this todo?

Reply:
per offline convo with @CodingCat, we need to figure out a way to move this one time sync before loadcheckpoint. 
![image](https://user-images.githubusercontent.com/1184534/64672167-8c093880-d41f-11e9-8ae9-37c50a789b41.png)
https://github.com/dmlc/rabit/blob/master/README.md

More detailed reason is we removed is_boostrap flag and use checkpoint load as flag to differentiate cache v.s seqno based allreduce results recovery https://github.com/dmlc/rabit/commit/9a7ac85d7eb65b1a0b904e1fa8d5a01b910adda4